### DISCUSSION TITLE
Make non-essential build steps soft fail

### DISCUSSION THREAD

**Discussion by non-descriptive** [helix-editor/helix#10086] [General] [answered]
I've tried to install helix via
```
cargo install --git https://github.com/helix-editor/helix  helix-term --force
```
and it failed on custom build step, attempting to download tree sitter grammars. 

```
  --- stdout
  Fetching 188 grammars
  185 updated grammars
        ada                now on ba0894efa03beb70780156b91e28c716b7a4764d
...<many more trees>
        yuck               now on e3d91a3c65decdea467adebe4127b8366fa47919
        zig                now on 0d08703e4c3f426ec61695d7617415fff97029bd
  Failure 1/3: lpf Git command failed.
  Stdout:
  Stderr: fatal: unable to access 'https://gitlab.com/TheZoq2/tree-sitter-lpf/': SSL connection timeout

  Failure 2/3: blueprint Git command failed.
  Stdout:
  Stderr: fatal: unable to access 'https://gitlab.com/gabmus/tree-sitter-blueprint/': SSL connection timeout

  Failure 3/3: t32 Git command failed.
  Stdout:
  Stderr: fatal: unable to access 'https://gitlab.com/xasc/tree-sitter-t32/': SSL connection timeout

  --- stderr
  thread 'main' panicked at helix-term/build.rs:5:26:
  Failed to fetch tree-sitter grammars: 3 grammars failed to fetch
  ```
  
Yes, some addresses inaccessible from my network, but I don't care about those grammars anyway and would like to skip them. 
Since they are non-essential to main installation process and can be fetched later manually I think custom build step shouldn't halt installation process completely, but just notify about errors during installation.

**Comment by pascalkuthe** [comment, marked as answer]
You can disable grammar build by setting an environment variable. Silent failing one errors is never a good thing

**Reply by non-descriptive** [reply]
is it `HELIX_DISABLE_AUTO_GRAMMAR_BUILD`?

**Reply by non-descriptive** [reply]
Yes it was. Thanks. 
I didn't said it should be silent. It's neither report about possibility to use environment variable, nor skips failed fetches, which is not really essential for installation - many other were fetched and can be build.
